### ISSUE TITLE
[Bug Report] Pid matching misfunctions when running dae inside a pid-namespaced container

### ISSUE THREAD

**Issue by jschwinger233** [daeuniverse/dae#384]
### Checks

- [X] I have searched the existing issues
- [X] I have read the documentation
- [X] Is it your first time sumbitting an issue

### Current Behavior

Running dae inside a container with an independent pid namespace leads to infinite loop. This is because cgroup/* bpf programs record host pid while dae setup namespaced pid by `os.GetPid()`.

Likely it's not a bug, as I am not aware of any solution except:
1. we match process name instead of pid
2. we ditch the pid matching to be in favor of mark matching (to distinguish dae traffic)

Still, the general pid routing fails to work properly when inside a pid-namespaced container.

Not saying we must take care of this, just open an issue for record and awareness.

### Steps to Reproduce

1. Prepare the conf:

```
global {
    tproxy_port: 12345
    log_level: trace
    wan_interface: auto
    allow_insecure: false
}

node {
    local: 'socks5://socks5:1080'
}

group {
    proxy {
        policy: min_moving_avg
    }
}

routing {
    fallback: proxy
}
```

2. Run dae inside a pid-namespaced container

```
# assume we have "dae" binary under pwd
docker run -td --name dae --privileged -v $(pwd):/host ubuntu:22.04 bash
docker exec dae mount bpffs -t bpf /sys/fs/bpf
docker exec dae mount -t debugfs none /sys/kernel/debug/
docker exec -it docker exec -it dae /host/dae run -c /host/conf.dae
```

3. dae gets stuck into infinite loop

### Environment

- **Dae version (use `dae --version`)**: dae version unstable-20231221.r627.84c8eeb
- **OS (e.g `cat /etc/os-release`)**: Ubuntu 22.04.3 LTS
- **Kernel (e.g. `uname -a`)**: Linux gray-Latitude-5530 6.2.0-39-generic #40~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 16 10:53:04 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
- **Others**: None

**Comment by mzz2017** [comment]
![image](https://github.com/daeuniverse/dae/assets/30586220/28b5eb0a-b76f-4f36-9ccf-81b6549f4baf)

https://hub.docker.com/r/daeuniverse/dae

Yes. It is what we recommend.

**Comment by jschwinger233** [comment]
@sumire88 Not really, this issue if for record and awarenes in case other users hit it.

**Comment by artiga033** [comment]
Friendly ping, if using dae in WSL on Windows, this also happens. Because in WSL all distro are actually namespace-isolated "containers". There is a "system distro" which is like the host machine in container context., entering via `wsl --debug-shell`. The BPF program gets pid from that system distro, but dae control plane is never expected being running there...

An workaround is add pname(dae) -> direct, but I do wish we can have a fundamental solution...

**Comment by olicesx** [comment]
Closing as outdated: reported against a pre-v2.0 build/environment that has since been superseded by the v2.0.0 / #1099 datapath and control-plane rework, and no longer reproducible against current main. If this still reproduces on the latest release, please open a new issue with current-version logs. Housekeeping after #1099.

**Comment by olicesx** [comment]
Closing as outdated: reported against a pre-v2.0 build/environment that has since been superseded by the v2.0.0 / #1099 datapath and control-plane rework, and no longer reproducible against current main. If this still reproduces on the latest release, please open a new issue with current-version logs. Housekeeping after #1099.
